apps/ejabberd/src/mod_commands.erl

```diff
@@ -107,25 +107,27 @@ commands() ->
       {result, ok}
      ],
      [
-      {name, get_last_messages},
-      {category, <<"messages">>},
-      {desc, <<"Get n last messages">>},
-      {module, ?MODULE},
-      {function, get_recent_messages},
-      {action, read},
-      {security_policy, [user]},
-      {args, [{caller, binary}, {other, binary}, {limit, integer}]},
-      {result, []}
+     {name, get_last_messages_with_everybody},
+     {category, <<"messages">>},
+     {desc, <<"Get n last messages with given contact, optionally before a certain date (unixtime)">>},
+     {module, ?MODULE},
+     {function, get_recent_messages},
+     {action, read},
+     {security_policy, [user]},
+     {args, [{caller, binary}]},
+     {optargs, [{before, integer, 0}, {limit, integer, 100}]},
+     {result, []}
      ],
      [
-      {name, get_messages},
+      {name, get_last_messages},
       {category, <<"messages">>},
-      {desc, <<"Get messages before a certain date ('before' is a unix timestamp in seconds)">>},
+      {desc, <<"Get n last messages with given contact, optionally before a certain date (unixtime)">>},
       {module, ?MODULE},
       {function, get_recent_messages},
       {action, read},
       {security_policy, [user]},
-      {args, [{caller, binary}, {other, binary}, {before, integer}, {limit, integer}]},
+      {args, [{caller, binary}, {other, binary}]},
```

Comment:
We can play with all names except 'caller', it has a special meaning in commands engine. We can change it to anything else, but it has to have the same name in all commands.